File: src/client/java/com/myudog/myulib/client/api/ui/component/ComputedTransform.java
package com.myudog.myulib.client.api.ui.component;

import com.myudog.myulib.api.core.ecs.IComponent;

public class ComputedTransform implements IComponent {
    public float x;
    public float y;
    public float width;
    public float height;
    public float scaleX = 1.0f;
    public float scaleY = 1.0f;
    public float rotation;
    public float opacity = 1.0f;
}

File: src/client/kotlin/com/myudog/myulib/client/api/ui/component/ComputedTransform.kt
package com.myudog.myulib.client.api.ui.component

import com.myudog.myulib.api.ecs.Component

/**
 * [API - ReadOnly] 存放佈局引擎計算後的絕對像素結果。
 * ⚠️ 注意：此組件由 LayoutSystem 自動更新，請勿手動修改其數值，否則會被覆蓋。
 */
data class ComputedTransform(
    var x: Float = 0f,
    var y: Float = 0f,
    var w: Float = 0f,
    var h: Float = 0f
) : Component